### PR TITLE
Refactor L006 & L039, introduce L048

### LINKED ISSUE #360: Lint rule for spaces surrounding apostrophes (')
Can we have a lint rule that checks for spaces surrounding apostrophes? Following Simon Holywell's SQL Style Guide:

> Although not exhaustive always include spaces:
> * ...
> * surrounding apostrophes (') where not within parentheses or with a trailing comma or semicolon.

reference: https://www.sqlstyle.guide/#white-space

### PR DESCRIPTION
**Pull request by alanmcruickshank**
This resolves #360 .

It also refactors L006. Since the introduction of L039 (excessive whitespace), L006 no longer needs to lint for _too much_ whitespace. This PR applies that fix and also make the logic more reusable so that it can be used for L048 (missing whitespace around quoted literal). The refactor also improves the error message to be more explanatory (e.g. "Missing whitespace before *").

In implementing these I noted that L039 was only flagging on the first issue found and not all occurrences. This PR also extends that rule to fix that issue.

NB: L006 and L048 check for whitespace before *and* after (as before) but the anchor point for both is now the same. This means in comparing sets of check_tuples it appears as though the same error has appeared at the same position and therefore only once. This is just an effect of comparing `set` objects. In real use, the effect is still the same except that the position of the second linting issue has been moved back by one. This feels like a more appropriate positioning for the error.

### CODE REVIEW

File: src/sqlfluff/core/rules/std/L006.py [hunk 1]
```diff
@@ -30,113 +31,187 @@ class Rule_L006(BaseRule):
...
+                                        raw=" ", pos_marker=sub_seg.pos_marker
+                                    ),
+                                )
+                            ],
                         )
-                        if anchor:
-                            description = "Operators should be followed by a space."
+                    )
+
+            if check_after:
```
Comment (barrywhart) [inline src/sqlfluff/core/rules/std/L006.py:161]: Nitpick: The `after` section of this code looks like a mirror image of the `_before` section. Would it be worthwhile to try and have one version of this code that can be run in both directions (forward and backward)? (I'm torn on this myself.)

File: test/fixtures/rules/std_rule_cases/L048.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+rule: L048
+
+test_pass_1:
+  # Test that we don't fail * operators in brackets
```
Comment (barrywhart) [inline test/fixtures/rules/std_rule_cases/L048.yml:4]: I don't understand the comment here -- it doesn't seem to relate to the test query.

File: test/fixtures/rules/std_rule_cases/L048.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  pass_str: "SELECT 'foo'"
+
+test_pass_2:
```
Comment (barrywhart) [inline test/fixtures/rules/std_rule_cases/L048.yml:7]: Isn't this a duplicate of `test_pass_1`?

File: src/sqlfluff/core/rules/std/L006.py [hunk 4]
```diff
@@ -1,6 +1,8 @@
 """Implementation of Rule L006."""
 
 
+from typing import Tuple, List
```
Comment (barrywhart) [inline src/sqlfluff/core/rules/std/L006.py:4]: I don't see any corresponding changes to the test file, `L006.yml`. Should there be new or updated tests to go with this?

File: src/sqlfluff/core/rules/std/L039.py [hunk 5]
```diff
@@ -32,6 +32,7 @@ def _eval(self, segment, parent_stack, **kwargs):
         # For the given segment, lint whitespace directly within it.
         prev_newline = True
         prev_whitespace = None
+        violations = []
```
Comment (barrywhart) [inline src/sqlfluff/core/rules/std/L039.py:35]: I don't see any corresponding changes to the test file, `L039.yml`. Should there be new or updated tests to go with this?

File: test/fixtures/rules/std_rule_cases/L048.yml (hunk 2 above)
Comment (alanmcruickshank) [reply test/fixtures/rules/std_rule_cases/L048.yml:4]: oops yeah, that's a remnant from L006

File: src/sqlfluff/core/rules/std/L006.py (hunk 4 above)
Comment (alanmcruickshank) [reply src/sqlfluff/core/rules/std/L006.py:4]: All the existing tests are still totally relevant. I take your point on maybe adding some extra tests though.

File: src/sqlfluff/core/rules/std/L039.py (hunk 5 above)
Comment (alanmcruickshank) [reply src/sqlfluff/core/rules/std/L039.py:35]: Nope. Existing tests are fine. The new changes will just make it faster.

File: src/sqlfluff/core/rules/std/L006.py (hunk 1 above)
Comment (alanmcruickshank) [reply src/sqlfluff/core/rules/std/L006.py:168]: I don't know if we can have the same code, but I've separated out two slightly more generic methods to reuse a chunk of the logic. That should at least minimize issues from repeated code. What do you think of it now?

**Pull request merged**